File: 682. Baseball Game.java


public int calPoints(String[] ops){
	// int[] re = New int[]{}
	int sum = 0;
	List<Integer> l = new LinkedList<>(); 
	for (String str:ops){
		if (str.equals("C")){
			sum -= l.removeLast()
		}
		else if (str.equals("D")){
			l.add(l.peekLast()*2);
			sum += l.peekLast()
		}
		else if (op.equals("+")) {
            l.add(l.peekLast() + l.get(l.size() - 2));
            sum += l.peekLast();
        }
        else {
            l.add(Integer.parseInt(op));
            sum += l.peekLast();
        }
	}
	return sum;

}

File: 682. Baseball Game.py
class Solution(object):
    def calPoints(self, ops):
        """
        :type ops: List[str]
        :rtype: int
        """
        valid=[]
        for o in ops:
            if o == "+":
                valid.append(valid[-1]+valid[-2])
            elif o == "C":
                valid.remove(valid[-1])
            elif o == "D":
                valid.append(2*valid[-1])
            else:
                valid.append(int(o))
                
        return sum(valid)
        